v22.1/api-support-policy.md

```diff
@@ -0,0 +1,109 @@
+toc: true
+docs_area: reference
+---
+
+Cockroach Labs exposes various application programming interfaces (APIs).
+
+The vast majority of changes to these interfaces are seamless additions of new functionality. However, some changes are backward-incompatible and may require you to adjust your integration. Changes to an API are introduced according to its *stability guarantees*, which depend on its *support policy*.
+
+This page includes the following information:
+
+- Our [support policies](#support-policies) for each type of API.
+- Our [stability guarantees](#stability-guarantees) for supported APIs.
+- A summary of [APIs](#apis) that CockroachDB makes available.
+
+## Support policies
+
+| Type             | Description                                                               | Stability guarantees |
+|------------------|---------------------------------------------------------------------------|----------------------|
+| Programmable     | Supported for interfacing with third-party automated tools.               | ✓                    |
+| Non-programmable | Supported for consumption by humans. Not supported for automation.        |                      |
+| Reserved         | Intended for use by CockroachDB developers. Not supported for public use. |                      |
+
+### Stability guarantees
+
+An API is *stable* if it does not receive backward-incompatible changes. API stability implies programmability.
+
+The following stability guarantees apply to **programmable** APIs. These APIs remain stable within major versions:
+
+- Backward-incompatible changes may be introduced in new major versions.
+- Backward-compatible changes may be introduced in new minor versions.
+
+Stability guarantees do **not** apply to non-programmable and reserved APIs:
+
+- Backward-incompatible changes may be introduced in new minor and major versions.
+
+#### API changes
+
+The following types of API changes qualify as *backward-incompatible* (or "breaking changes"):
+
+- Removal of an endpoint, SQL statement, [built-in function](functions-and-operators.html#built-in-functions), [cluster setting](cluster-settings.html), or session variable.
+- Addition or removal of a mandatory command-line flag.
+- Change in SQL statement syntax, HTTP request format, HTTP response format, or structured log format.
+- Change in data type, default value, or behavior of a [built-in function](functions-and-operators.html#built-in-functions), [cluster setting](cluster-settings.html), or session variable.
+- Change to [authorization](security-reference/authorization.html) requirements.
+
+The following types of API changes qualify as *backward-compatible*. This list is not exhaustive.
+
+- Removal or change of any functionality documented or named as "experimental", "beta", or otherwise not fully supported.
+- Deprecation of any functionality without removal.
+- Addition or removal of an optional command-line flag.
```

Comment:
I would consider removal of a command-line flag a breaking change, whether or not it's optional. Even optional flags could have material changes to the operation of a CLI.